shotover-proxy/src/protocols/cassandra_protocol2.rs

```diff
@@ -577,12 +584,53 @@ impl Decoder for CassandraCodec2 {
         &mut self,
         src: &mut BytesMut,
     ) -> std::result::Result<Option<Self::Item>, Self::Error> {
+        // if we need to close the connection return an error.
+        if self.force_close.is_some() {
```

Comment:
We can simplify this:
```rust
        if let Some(result) = self.force_close.take() {
            debug!("Closing errored connection: {:?}", &result);
            return Err(anyhow::Error::msg(result));
        }
```

Reply:
Is waiting for the next message to come in before closing the connection a hack? or just how the protocol works?

Reply:
> If we shut the connection before the client receives the message the client will receive a peer closed the connection message and may try again with the same invalid protocol and we can try to process it again.

Ahh, makes sense, this explains away my concern.
Might be worth adding as a comment.